yazi-core/src/manager/watcher.rs

```diff
@@ -72,7 +72,7 @@ impl Watcher {
 				(old.difference(new).cloned().collect(), new.difference(old).cloned().collect())
 			};
 
-			to_unwatch.retain(|u| watcher.unwatch(u).is_ok());
+			to_unwatch.retain(|u| watcher.unwatch(u).is_ok() || !u.exists());
```

Comment:
`.unwatch()` should already return an error when it fails to unwatch a non-existent directory, why don't we just use it directly?

Reply:
you mean to analyze the cause of the error before deciding whether to remove the path?

Reply:
Yep, I'll fix it now